pkg/controller/common/license/watches.go

```diff
@@ -7,31 +7,29 @@ package license
 import (
 	"github.com/elastic/cloud-on-k8s/pkg/controller/common"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/common/watches"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-func NewWatch(fn handler.Mapper) watches.HandlerRegistration {
+func NewWatch(fn handler.MapFunc) watches.HandlerRegistration {
```

Comment:
I believe this was used originally in the remotecluster controller to trigger a reconciliation if the license changed. E.g. if a user added an Enterprise license it would run the controller and setup the remote cluster relationship if the user had configured it. We should probably open an issue to check if this is still needed.

Reply:
As the code seems to be unused, I'll delete it.

Reply:
I would keep it until we understand if something has been missed: https://github.com/elastic/cloud-on-k8s/issues/4152